naps/cores/video/gamma_corrector_test.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+from os.path import join, dirname
+import imageio
+from nmigen import *
+from nmigen.sim import Passive
+from naps import SimPlatform
+from naps.stream import write_to_stream
+from naps.cores.video import ImageStream, TableGammaCorrector, write_frame_to_stream, read_frame_from_stream
+
+
+class GammaCorrectorTest(unittest.TestCase):
+    def check_output(self, corrector_gen, gamma):
+        platform = SimPlatform()
+        m = Module()
+
+        input = ImageStream(8)
+        transformer = m.submodules.transformer = corrector_gen(input, gamma)
+        image = imageio.imread(join(dirname(__file__), "wavelet", "che_32.png"))
+
+        # correct the image ourselves to check the corrector's work
+        if gamma != 1:
+            bpp = 8
+            max_pix = 2**bpp - 1
+            lut = list(int(max_pix*((v/max_pix)**gamma)) for v in range(max_pix+1))
+            image_corrected = list(list(lut[pixel] for pixel in line) for line in image)
+        else: # gamma = 1 should not change image at all
+            image_corrected = list(list(pixel for pixel in line) for line in image)
+
+        def write_process():
+            yield from write_frame_to_stream(input, image, pause=False)
+            yield Passive()
+            while True:
```

Comment:
This `while` loop does not seem necessary.

Reply:
I copied it from `debayer_test.py` because I figured it was needed to compensate for the latency of the core, but it seems it is necessary because of how the `ImageConvoluter` works, which I do not use. So I can remove it.